### PR TITLE
Add VSCode keymap for Windows

### PR DESCRIPTION
**Pull request by Huliiiiii**
Release Notes:

- N/A

### CODE REVIEW

File: crates/settings/src/settings.rs [hunk 1]
```diff
@@ -35,9 +35,12 @@ pub fn default_settings() -> Cow<'static, str> {
 #[cfg(target_os = "macos")]
 pub const DEFAULT_KEYMAP_PATH: &str = "keymaps/default-macos.json";
 
-#[cfg(not(target_os = "macos"))]
+#[cfg(all(not(target_os = "macos"), not(target_os = "windows")))]
```
Comment (maxdeviant) [inline crates/settings/src/settings.rs:38]: This should probably just be used on Linux:

```suggestion
#[cfg(target_os = "linux")]
```

**Comment by notpeter** [comment]
Hi @Huliiiiii,

Thanks for doing the work on this.  In parallel I built out a linux keymap for VSCode, JetBrain, Atom and SublimeText (great minds).
- https://github.com/zed-industries/zed/pull/13792

The way I've configured it there isn't an explicit set of default Windows keybinds, just a MacOS and a non-MacOS (linux and windows).  Please test `main` and feel free to update this PR if there's shortcuts I missed or got wrong.

File: assets/keymaps/default-windows.json [hunk 2]
```diff
@@ -0,0 +1,681 @@
+[
+  // Standard Windows bindings
+  {
+    "bindings": {
```
Comment (mikayla-maki) [inline assets/keymaps/default-windows.json:4]: Is this keymap particularly different from the linux keymap? If we can use the linux keymap in both places, that would be preferable :)

**Review by mikayla-maki — CHANGES_REQUESTED** [review]

File: assets/keymaps/default-windows.json (hunk 2 above)
Comment (Huliiiiii) [reply assets/keymaps/default-windows.json:4]: | key | windows | linux 
| --- | --- | --- 
| shift-alt-up | editor::DuplicateLineUp | editor::AddSelectionAbove
| shift-alt-down | editor::DuplicateLineDown | editor::AddSelectionBelow
| shift-alt-f | editor::Format | null
| ctrl-alt-shift-up | editor::AddSelectionAbove | editor::DuplicateLineUp
| ctrl-alt-shift-down | editor::AddSelectionBelow | editor::DuplicateLineDown
| ctrl-f12 | editor::GoToImplementation | editor::GoToTypeDefinition
| shift-f12 | editor::GoToTypeDefinition | editor::GoToImplementation
| ctrl-shift-f12 | editor::GoToImplementationSplit | editor::GoToDefinitionSplit (It's  ctrl-shift-f10 in default-linux.json (typo?))

Comment (Huliiiiii) [reply assets/keymaps/default-windows.json:4]: I also set `ctrl-k` to null to override the default value `editor::CutToEndOfLine`, which ensures that `ctrl-k` key combinations won't accidentally cut.

**Comment by mikayla-maki** [comment]
Thank you for this work, however I don't think 8 changed keybindings is enough impetus for us to create a whole new key map with ~500 entries that we have to update for every change.

**Pull request closed without merging**